### PR TITLE
Date range validation not happening with CMS page's custom design update

### PR DESCRIPTION
**Pull request by maddumarala**
### Description (*)
In the process of creating CMS page, if user wants to  add custom design update, the from_date and to_date validation is not happening. Added a fix for that.

### Manual testing scenarios (*)

Go to Admin>content>pages>Add new page
Add required info
Add custom design update with from date(02/1/2019) and to date(01/20/2019)
No error message given.But in other places which uses this functionality validate the date range(Catalog price rules)

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=20868) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @maddumarala. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by slavvka** [comment]
Hey @maddumarala thank you for the contribution. Please consider covering the fix with automated tests. Thanks

**Comment by maddumarala** [comment]
> Hey @maddumarala thank you for the contribution. Please consider covering the fix with automated tests. Thanks

Hi @slavvka , Could you please explain on how to cover this fix with automated tests ? I thought since Travis CI still pending means testing is still running ?

**Comment by slavvka** [comment]
> > Hey @maddumarala thank you for the contribution. Please consider covering the fix with automated tests. Thanks
> 
> Hi @slavvka , Could you please explain on how to cover this fix with automated tests ? I thought since Travis CI still pending means testing is still running ?

Hey @maddumarala your are right. Travis tests are running. But you added additional validations which should be tested as well - they didn't exist previously and therefore weren't tested. In your case I would suggest adding an integration test to check the validation on the resource model level.

**Comment by sidolov** [comment]
@maddumarala , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

File: app/code/Magento/Cms/Model/ResourceModel/Page.php [hunk 1]
```diff
@@ -265,6 +272,33 @@ protected function isValidPageIdentifier(AbstractModel $object)
         return preg_match('/^[a-z0-9][a-z0-9_\/-]+(\.[a-z0-9_-]+)?$/', $object->getData('identifier'));
     }
 
+    /**
+     *  Validate date range
+     *
+     * @param AbstractModel $object
+     * @return bool
+     */
+    protected function validateDateRange(AbstractModel $object)
```
Comment (sivaschenko) [inline app/code/Magento/Cms/Model/ResourceModel/Page.php:281]: Please change the access level of this method to private and cover this validation with integration tests

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @maddumarala please see the review note. Let us know what help do you need to complete the pull request.

**Comment by sidolov** [comment]
@maddumarala , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**